lib/rubocop/cop/rspec/named_subject.rb

```diff
@@ -45,12 +45,14 @@ class NamedSubject < Cop
         MSG = 'Name your test subject if you need '\
               'to reference it explicitly.'
 
-        def_node_matcher :rspec_block?, <<-PATTERN
-          {
-            #{Examples::ALL.block_pattern}
-            #{Hooks::ALL.block_pattern}
-          }
-        PATTERN
+        def_runtime_node_matcher :rspec_block? do
+          <<-PATTERN
+            {
+              #{Examples::ALL.block_pattern}
+              #{hook_selectors.block_pattern}
```

Comment:
@marcandre summoning you to gather your opinion on a lazy node matcher definition.
We're experiencing a problem with hard-coded DSL definition (say, hooks are limited to `begin`/`after`/`around`), while RSpec allows for various aliases (examples, example groups) that we're unable to properly detect.

However, due to lazy config loading that happens after the cops are loaded and helper `Language` module is loaded, we resort to introduce a `def_runtime_node_matcher` that is passed a block to postpone node pattern evaluation.

Do you see any problem with that?

@andrykonchin summoning you as our de-facto performance expert. Do you think this change comes with a significant performance impact?

Reply:
`NodePattern` is more flexible than ever, especially with the named parameters that I introduced recently (which can be specified at definition time or when calling), so I doubt that it's needed. Performance-wise my guess is it would be somewhere between really slow and attrociously slow 🤣.
I can look at this PR in depth tomorrow if needed, but to be clear: don't use `def_runtime_node_matcher`...

Reply:
E.g. if the keywords `before/after/...` can only be known later...
```ruby
def_node_matcher :something, '(something (sym %keywords) ...', keywords: ->(keyword) { acceptable_keyword_set.include?(keyword) }
```

Just define `acceptable_keyword_set` to return a memoized `Set` of keywords...

Reply:
> between really slow and attrociously slow

😆 

Thanks a lot @marcandre! This is something I've totally overlooked, even with [my recent usage of this feature](https://github.com/rubocop-hq/rubocop-rspec/pull/930/files#diff-2aecb8a103588de3553f5448956d701fR59).

@sl4vr Would you like to give another approach a shot?

Reply:
I'm ashamed to admit I was referring [outdated docs](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/NodePattern/Macros) while working on this implementation 🤷‍♂️. Actually I've noticed that only after publishing this PR when I tried to investigate on failing tests.

@pirj Of course! I'll try to rework this PR in order to adopt `NodePattern`'s named parameters.

Reply:
Did you run into difficulties? I notice that `def_runtime_node_matcher` is still there